src/lib/blockchain/getWeb3.js

```diff
@@ -111,11 +111,45 @@ const setWallet = (rpcUrl, isHomeNetwork = false) =>
     this.setProvider(engine);
   };
 
+/**
+  Method to extend web3 to poll tx receipt mined. Usage:
+  web3.eth.getTransactionReceiptMined = _getTransactionReceiptMined;
+  Gist: https://gist.github.com/xavierlepretre/88682e871f4ad07be4534ae560692ee6
+**/
+const _getTransactionReceiptMined = function(txHash, interval){
```

Comment:
web3 1.0 already provides this functionality and will only resolve the promise once the tx is mined

Reply:
I tried some stuff with events but those did not fire. 
But removed and changed to `getTransactionReceipt`